e3db/src/main/java/com/tozny/e3db/Identity.kt

```diff
@@ -28,15 +28,16 @@ open class PartialIdentityClient @JvmOverloads constructor(val client: Client, v
   }
 
   private fun validatePassword(currentPassword: String): Boolean {
-    val deriveNoteCreds = Realm.deriveNoteCreds(identityConfig.realmName, identityConfig.username, currentPassword, CredentialType.PASSWORD)
+    // identityConfig.username non-null asserted due to IdentityConfig providing username as an optional field
+    val deriveNoteCreds = Realm.deriveNoteCreds(identityConfig.realmName, identityConfig.username!!, currentPassword, CredentialType.PASSWORD)
```

Comment:
`!!` should only be used in places where there is a guarantee that the optional value is present. If know that than this is ok. If we don't know that than we should capture that and return an error value of some kind